crates/router/src/types/api.rs

```diff
@@ -39,10 +39,12 @@ pub trait ConnectorCommon {
     fn id(&self) -> &'static str;
 
     /// HTTP header used for authorization.
+    #[allow(clippy::type_complexity)]
```

Comment:
Don't allow clippy warnings, if the warning can't be resolve or takes much Dev-effort then we can allow, this didn't look like one to allow.